test/puppetlabs/puppetdb/integration/smoke.clj

```diff
@@ -27,3 +28,94 @@
+     {:name "literal_float" :code "3.14" :expected 3.14}
+     {:name "literal_true" :code "true" :expected true}
+     {:name "literal_string" :code "\"hello\"" :expected "hello"}
+     {:name "string_with_single_quote" :code "\"ta'phoenix\"" :expected "ta'phoenix"}
+     {:name "string_with_double_quote" :code "\"he said \\\"hi\\\"\"" :expected "he said \"hi\""}
+     {:name "regexp" :code "/[a-z]+/" :expected "/[a-z]+/"}
+     {:name "deferred" :code "Deferred('join', [[1, 2, 3], ':'])" :expected "Deferred({'name' => 'join', 'arguments' => [[1, 2, 3], ':']})"}
+     {:name "sensitive_deferred" :code "Sensitive(Deferred('join', [[1, 2, 3], ':']))" :expected nil}
+     {:name "sensitive" :code "Sensitive('password')" :expected nil}
+     {:name "timestamp" :code "Timestamp('2012-10-10')" :expected "2012-10-10T00:00:00.000000000 UTC"}
+     {:name "hash_and_array" :code "{'a' => [1, 2, 3], 'b' => 'hello'}" :expected {:a [1, 2, 3] :b "hello"} }
+     {:name "special_key" :code "{ ['special', 'key'] => 10 }" :expected {(keyword "[\"special\", \"key\"]") 10}}
+     {:name "hash_with_sensitive_key" :code "{Sensitive(hush) => 42 }"
+      :valid? (fn [test-data]
+                (let [test-keys (keys test-data)]
+                  (and (= (count test-keys) 1)
+                       (re-matches #"#<Sensitive \[value redacted\]:[0-9]+>" (name (first test-keys)))
+                       (= 42 ((first test-keys) test-data)))))}
+     {:name "hash_with_sensitive_val" :code "{ x => Sensitive(hush) }"
+      :valid? (fn [test-data]
+                (and (= (count (keys test-data)) 1)
+                     (re-matches #"#<Sensitive \[value redacted\]:[0-9]+>" (:x test-data))))}
+     {:name "hash_ptype_key" :code "{'__ptype' => 10}" :expected {(keyword "reserved key: __ptype") 10}}
+     {:name "binary_value" :code "Binary(\"hello\", \"%s\")" :expected "aGVsbG8="}
+     {:name "a_type" :code "Integer[0,100]" :expected "Integer[0, 100]"}
+
+     ; Car is defined in the manifest generated in the test below
+     ; {:name "user_defined_type" :code "Car" :expected "Car"}
+     ; {:name "user_defined_object" :code "Car(abc123)" :expected "Car({'regnbr' => 'abc123'})"}
+     ])
+
+  (deftest ^:integration rich-data-agent-run
+    (with-open [pg (int/setup-postgres)
+                pdb (int/run-puppetdb pg {})
+                ps (int/run-puppet-server [pdb] {:agent {:rich-data true}})]
+
+      (testing "Agent run succeeds"
+        (let [manifest (str "type Car = Object[attributes => {regnbr => String}];\n"
+                            (apply str (for [{:keys [name code]} rich-data-tests]
+                                         (str "notify { '" name "':"
+                                              "  message => " code
+                                              "}"))))
+              {:keys [out]} (int/run-puppet-as "my-agent" ps pdb manifest {:rich-data true})]
+          (is (re-find #"a-z" out))))
```

Comment:
This is actually looking for the three literal characters in sequence "a-z"?

Reply:
Ya, that's left over from when the test only used Regex, so the notify was "/[a-z]+/" and that made sure that it had been notified to the stdout of the puppet run